torch/utils/_mode_utils.py

```
@@ -12,4 +13,46 @@ def all_same_mode(modes):
+    if not modes:
+        return False
+    
+    mode_type = type(modes[0])
+    mode_count = len(modes)
+    
+    # Use cached validation
+    if not _validate_mode_consistency(mode_type, mode_count):
+        return False
+    
+    # Check all modes are same type
+    return all(type(mode) == mode_type for mode in modes)
```

Ruff (0.14.6)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)